### DISCUSSION TITLE
Upscaling and Ram management

### DISCUSSION THREAD

**Discussion by GeekyG4mer** [rafaelvcaetano/melonDS-android#299] [Ideas]
Can upscaling be implemented. Or maybe make the apl a little less ram intensive, I have a Snapdragon 730G processor and this emu consumes more ram than a 3ds emulator. It would be great to get either of these looked into.

**Comment by rafaelvcaetano** [comment]
Upsacling can be implemented after the OpenGL renderer is implemented. It's planned but it doesn't have a high priority.
Regarding RAM consumption, I've seen it being a problem. What levels of RAM usage are you seeing? An average, the RAM usage should be the size of the ROM + some 10 MB for the emulator and 60 to 100 MB for the frontend.

**Reply by GeekyG4mer** [reply]
I tested the Ram consumption, it was just as you said but sometimes the emulator can go up to 6 to 700 megabytes of ram while the rom being of 150 mb.
I really hope that upscaling would be implemented ,at least 2x, as this the only thing that is lacking in comparison to drastic

**Reply by rafaelvcaetano** [reply]
Alright. Thanks for letting me know. I'll see what's going on. Does this happen immediately after loading the first ROM? Or after launching several ROMs (or the same one several times)?

**Reply by GeekyG4mer** [reply]
Not quite sure
